src/adera/machines/links.cpp

```diff
@@ -1,12 +1,63 @@
 #include "links.h"
 
+using namespace osp;
+
 using osp::link::MachTypeReg_t;
 using osp::link::MachTypeId;
 
 namespace adera
 {
 
-MachTypeId const gc_mtUserCtrl = osp::link::MachTypeReg_t::create();
-MachTypeId const gc_mtMagicRocket = osp::link::MachTypeReg_t::create();
+MachTypeId const gc_mtUserCtrl      = MachTypeReg_t::create();
+MachTypeId const gc_mtMagicRocket   = MachTypeReg_t::create();
+MachTypeId const gc_mtRcsDriver     = MachTypeReg_t::create();
+
+float thruster_influence(Vector3 const pos, Vector3 const dir, Vector3 const cmdLin, Vector3 const cmdAng) noexcept
+{
+    using Magnum::Math::cross;
+    using Magnum::Math::dot;
+
+    // Thrust is applied in opposite direction of thruster nozzle direction
+    Vector3 const thrustDir = std::negate{}(dir.normalized());
+
+    float angInfluence = 0.0f;
+    if (cmdAng.dot() > 0.0f)
+    {
+        Vector3 const torque = cross(pos, thrustDir).normalized();
+        angInfluence = dot(torque, cmdAng.normalized());
+    }
+
+    float linInfluence = 0.0f;
```

Comment:
minor: You could use a trinary conditional on these so that linInfluence is const.